js/src/common/Model.ts

```diff
@@ -111,6 +112,19 @@ export default abstract class Model {
 
     if ('attributes' in data) {
       this.data.attributes ||= {};
+
+      // @deprecated
+      // Filter out relationships that got in by accident.
+      for (const key in data.attributes) {
+        const val = data.attributes[key];
+        if (val && val instanceof Model) {
+          fireDebugWarning('Model.pushData() was passed a Model instance in the `attributes` field. This is not supported.');
```

Comment:
Could we use the deprecation warning helper here instead, and add a message to the top of the PR explaining how this should be done instead? (e.g. https://github.com/flarum/core/pull/3212)

```suggestion
          fireDeprecationWarning('Providing models as attributes to `Model.pushData()` or `Model.pushAttributes()` is deprecated.', '3249');
```